### PR TITLE
Upgrade to webpack 2.2

### PR DESCRIPTION
**Pull request by mikenikles**
Now that the final 2.2 release of webpack has [officially been released](https://medium.com/webpack/webpack-2-2-the-final-release-76c3d43bf144#.5y82q2fwm), this PR upgrades us to it. In future PRs, we will start to take advantage of some new features.

For now, we already benefit from a few improvements, such as the following console output to indicate big chunks when we build our production bundle.:

image

## Changes
- Upgraded to webpack 2.2
- Replaced the `NoErrorPlugin` with the `NoEmitOnErrorsPlugin`

## How to test-drive this PR
- Run `npm run dev` and preview Merlin's Potions.
  - Verify that the site works as expected.
- Run `npm run prod:build`
  - Verify that the production bundle gets built correctly.

### CODE REVIEW

**Comment by mikenikles** [comment]
@jeffkamo do you think we can shrink the size of the 3 carousel images to get rid of that warning in the console output?

File: package.json [hunk 1]
```diff
@@ -85,7 +85,7 @@
     "svgo": "0.7.1",
     "text-loader": "0.0.1",
     "validator": "6.2.0",
-    "webpack": "2.1.0-beta.27",
+    "webpack": "2.2",
```
Comment (DanielBrookRoberge) [inline package.json]: Should this be `"2.2.0"` to make sure we're locked to a specific version?

**Comment by DanielBrookRoberge** [comment]
One minor comment, then :+1: when that's fixed.

**Comment by jeffkamo** [comment]
@mikenikles Absolutely! I'll see what I can do

**Pull request merged**